### PR TITLE
Update roadmap for v0.3.0

### PR DESCRIPTION
**Pull request by arkodg**
Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

**Comment by arkodg** [comment]
ptal @danehans @LukeShu @AliceProxy @Xunzhuo @skriss @youngnick

**Review by AliceProxy — DISMISSED** [review]
Looks fine to me

**Review by Xunzhuo — APPROVED** [review]
🚀

File: docs/latest/design/roadmap.md
Comment (danehans) [inline docs/latest/design/roadmap.md:40]: IMHO this is overpromising. I would like to understand which maintainers will support these features. We should not depend on the community to deliver roadmap features. I'm in the process of updating #336, but this only covers the AuthN design. From recent feedback (#675 #677) and past experience, the AuthN design alone will take time. Since AuthN is laying the foundation for our approach to Gateway API extensions, this may cause a delay in starting #670. What was the conclusion for delaying RL for RLQS?

File: docs/latest/design/roadmap.md
Comment (danehans) [inline docs/latest/design/roadmap.md:44]: I suggest moving some of the v0.3 deliverables to v0.4. Note that the original v0.3 deliverables were based on a 6-month release cycle.

**Review by danehans — CHANGES_REQUESTED** [review]

File: docs/latest/design/roadmap.md
Comment (arkodg) [reply docs/latest/design/roadmap.md:40]: @danehans I assigned #670 to myself, I can design many aspects of it while waiting on the extension decision.
reg RLQS
* we dont have any info on when it will release
* will it be a better fit than Envoy's native RLS for the Gateway case ?
* will it be more stable  ?
so will start the initial design with Envoy RLS, and can raise a follow up issue to consider moving to RLQS when we
have enough data to answer above questions

File: docs/latest/design/roadmap.md
Comment (skriss) [inline docs/latest/design/roadmap.md:35]: Related to @danehans's comment, maybe worth calling out the deliverable of establishing a pattern for all extensions, since this is a big part of what we're actually doing with the authn design.

**Review by skriss — APPROVED** [review]
one suggestion, but LGTM overall

File: docs/latest/design/roadmap.md
Comment (arkodg) [reply docs/latest/design/roadmap.md:35]: Good point, will add

File: docs/latest/design/roadmap.md
Comment (Xunzhuo) [inline docs/latest/design/roadmap.md:37]: Can we link this to a specific issue like other goals in v0.3.0? This will be more clear for a roadmap.

Comment (danehans) [reply docs/latest/design/roadmap.md:37]: It should be linked to https://github.com/envoyproxy/gateway/pull/529 and  https://github.com/envoyproxy/gateway/issues/670. IMO https://github.com/envoyproxy/gateway/issues/671 is a stretch for v0.3, so feel free to add it to v0.4 if you feel confident in delivering the feature.

Comment (arkodg) [reply docs/latest/design/roadmap.md:37]: I think @Xunzhuo was referring to create a specific epic, created this one https://github.com/envoyproxy/gateway/issues/707 and will link it

**Comment by danehans** [comment]
Codecov upload [failure](https://github.com/envoyproxy/gateway/actions/runs/3413608671/jobs/5680572431), rerunning CI.

File: docs/latest/design/roadmap.md
Comment (Xunzhuo) [reply docs/latest/design/roadmap.md:45]: This is my concern too.

**Review by Xunzhuo — COMMENTED** [review]
https://github.com/envoyproxy/gateway/issues/657 https://github.com/envoyproxy/gateway/issues/656 https://github.com/envoyproxy/gateway/issues/564

BTW, need we add these three high-priority filters implementations into roadmap?

File: docs/latest/design/roadmap.md
Comment (danehans) [inline docs/latest/design/roadmap.md:41]: IMO we're still too aggressive with our v0.3 deliverables. For example, #642 should not be on the roadmap until someone is assigned and has confirmed delivering the feature for the milestone.

**Comment by arkodg** [comment]
> #656

@Xunzhuo these have been updated in the top level epic https://github.com/envoyproxy/gateway/issues/707

**Review by Xunzhuo — APPROVED** [review]
LGTM:)

**Review by danehans — APPROVED** [review]
Since a majority of maintainers have approved, I will not block. I am concerned that we are overcommitted for v0.3 but we can update the roadmap if needed.

**Pull request merged**
